src/jit.c

```diff
@@ -0,0 +1,1550 @@
+    R28,
+    R29,
+    R30,
+    SP,
+    RZ = 31
+};
+
+enum Opcode {
+    /* AddSubOpcode */
+    AS_ADD = 0,
+    AS_SUB = 2,
+    AS_SUBS = 3,
+    /* LogicalOpcode */
+    LOG_AND = 0x00000000U,  // 0000_0000_0000_0000_0000_0000_0000_0000
+    LOG_ORR = 0x20000000U,  // 0010_0000_0000_0000_0000_0000_0000_0000
+    LOG_ORN = 0x20200000U,  // 0010_0000_0010_0000_0000_0000_0000_0000
+    LOG_EOR = 0x40000000U,  // 0100_0000_0000_0000_0000_0000_0000_0000
+    /* LoadStoreOpcode */
+    LS_STRB = 0x00000000U,    // 0000_0000_0000_0000_0000_0000_0000_0000
+    LS_LDRB = 0x00400000U,    // 0000_0000_0100_0000_0000_0000_0000_0000
+    LS_LDRSBW = 0x00c00000U,  // 0000_0000_1100_0000_0000_0000_0000_0000
+    LS_STRH = 0x40000000U,    // 0100_0000_0000_0000_0000_0000_0000_0000
+    LS_LDRH = 0x40400000U,    // 0100_0000_0100_0000_0000_0000_0000_0000
+    LS_LDRSHW = 0x40c00000U,  // 0100_0000_1100_0000_0000_0000_0000_0000
+    LS_STRW = 0x80000000U,    // 1000_0000_0000_0000_0000_0000_0000_0000
+    LS_LDRW = 0x80400000U,    // 1000_0000_0100_0000_0000_0000_0000_0000
+    LS_STRX = 0xc0000000U,    // 1100_0000_0000_0000_0000_0000_0000_0000
+    LS_LDRX = 0xc0400000U,    // 1100_0000_0100_0000_0000_0000_0000_0000
+    /* LoadStorePairOpcode */
+    LSP_STPX = 0xa9000000U,  // 1010_1001_0000_0000_0000_0000_0000_0000
+    LSP_LDPX = 0xa9400000U,  // 1010_1001_0100_0000_0000_0000_0000_0000
+    /* UnconditionalBranchOpcode */
+    BR_BR = 0xd61f0000U,   // 1101_0110_0001_1111_0000_0000_0000_0000
+    BR_BLR = 0xd63f0000U,  // 1101_0110_0011_1111_0000_0000_0000_0000
+    BR_RET = 0xd65f0000U,  // 1101_0110_0101_1111_0000_0000_0000_0000
+    /* UnconditionalBranchImmediateOpcode */
+    UBR_B = 0x14000000U,  // 0001_0100_0000_0000_0000_0000_0000_0000
+    /* ConditionalBranchImmediateOpcode */
+    BR_Bcond = 0x54000000U,
+    /* DP2Opcode */
+    DP2_UDIV = 0x1ac00800U,  // 0001_1010_1100_0000_0000_1000_0000_0000
+    DP2_LSLV = 0x1ac02000U,  // 0001_1010_1100_0000_0010_0000_0000_0000
+    DP2_LSRV = 0x1ac02400U,  // 0001_1010_1100_0000_0010_0100_0000_0000
+    DP2_ASRV = 0x1ac02800U,  // 0001_1010_1100_0000_0010_1000_0000_0000
+    /* DP3Opcode */
+    DP3_MADD = 0x1b000000U,  // 0001_1011_0000_0000_0000_0000_0000_0000
+    DP3_MSUB = 0x1b008000U,  // 0001_1011_0000_0000_1000_0000_0000_0000
+    /* MoveWideOpcode */
+    MW_MOVN = 0x12800000U,  // 0001_0010_1000_0000_0000_0000_0000_0000
+    MW_MOVZ = 0x52800000U,  // 0101_0010_1000_0000_0000_0000_0000_0000
+    MW_MOVK = 0x72800000U,  // 0111_0010_1000_0000_0000_0000_0000_0000
+};
+
+enum Condition {
+    COND_EQ,
+    COND_NE,
+    COND_HS,
+    COND_LO,
+    COND_GE = 10,
+    COND_LT = 11,
+};
+#endif
+
+enum VM_REG {
+    VM_REG_0 = 0,
+    VM_REG_1,
+    VM_REG_2,
+    VM_REG_3,
+    VM_REG_4,
+    VM_REG_5,
+    VM_REG_6,
+    VM_REG_7,
+    VM_REG_8,
+    VM_REG_9,
+    VM_REG_10,
+    N_VM_REGS,
+};
+
+enum operand_size {
+    S8,
+    S16,
+    S32,
+};
+
+#define REGISTER_MAP_SIZE 11
```

Comment:
Instead of defining `REGISTER_MAP_SIZE` here, we can make use of macro `ARRAYS_SIZE` after defining `register_map`.